### PR TITLE
refactor: update `stats/base/sstdevyc` native addon from C++ to C

### PR DESCRIPTION
**Pull request by Vinit-Pandit**
Resolves none.

## Description

> What is the purpose of this pull request?

This pull request:

-  update `stats/base/sstdevyc` native addon from C++ to C

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves none

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/stats/base/sstdevyc/manifest.json [hunk 1]
```diff
@@ -24,6 +26,45 @@
...
+        "@stdlib/napi/argv-int64",
+        "@stdlib/napi/argv-strided-float32array",
+        "@stdlib/napi/create-double",
+        "@stdlib/napi/argv-float"
+      ]
+    },
+    {
+      "task": "benchmark",
+      "src": [
+        "./src/sstdevy.c"
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/stats/base/sstdevyc/manifest.json]:
```suggestion
        "./src/sstdevyc.c"
```

File: lib/node_modules/@stdlib/stats/base/sstdevyc/manifest.json [hunk 2]
```diff
@@ -24,6 +26,45 @@
   ],
   "confs": [
     {
+     "task": "build",
+      "src": [
+        "./src/sstdevy.c"
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/stats/base/sstdevyc/manifest.json]:
```suggestion
        "./src/sstdevyc.c"
```

File: lib/node_modules/@stdlib/stats/base/sstdevyc/src/addon.c [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
+	STDLIB_NAPI_ARGV_FLOAT( env, correction, argv, 1 );
+	STDLIB_NAPI_ARGV_INT64( env, stride, argv, 3 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, stride, argv, 2)
+	STDLIB_NAPI_CREATE_DOUBLE( env, stdlib_strided_sstdevyc( N, correction, X, stride ), v );
+	return v;
+}
+
+
+STDLIB_NAPI_MODULE_EXPORT_FCN( addon )
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/sstdevyc/src/addon.c]: Same issues. See other PRs.

File: lib/node_modules/@stdlib/stats/base/sstdevyc/src/addon.c [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+	STDLIB_NAPI_ARGV( env, info, argv, argc, 4 );
+	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
+	STDLIB_NAPI_ARGV_FLOAT( env, correction, argv, 1 );
+	STDLIB_NAPI_ARGV_INT64( env, stride, argv, 3 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, stride, argv, 2)
+	STDLIB_NAPI_CREATE_DOUBLE( env, stdlib_strided_sstdevyc( N, correction, X, stride ), v );
+	return v;
+}
+
+
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/stats/base/sstdevyc/src/addon.c]:
```suggestion
```

File: lib/node_modules/@stdlib/stats/base/sstdevyc/src/addon.c [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+* @param info   callback data
+* @return       Node-API value
+*/
+static napi_value addon( napi_env env, napi_callback_info info ) {
+	STDLIB_NAPI_ARGV( env, info, argv, argc, 4 );
+	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
+	STDLIB_NAPI_ARGV_FLOAT( env, correction, argv, 1 );
+	STDLIB_NAPI_ARGV_INT64( env, stride, argv, 3 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, stride, argv, 2)
+	STDLIB_NAPI_CREATE_DOUBLE( env, stdlib_strided_sstdevyc( N, correction, X, stride ), v );
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/stats/base/sstdevyc/src/addon.c]:
```suggestion
	STDLIB_NAPI_CREATE_DOUBLE( env, (double)stdlib_strided_sstdevyc( N, correction, X, stride ), v );
```

File: lib/node_modules/@stdlib/stats/base/sstdevyc/src/addon.c (hunk 3 above)
Comment (kgryte) [reply lib/node_modules/@stdlib/stats/base/sstdevyc/src/addon.c]: Still missing a trailing newline.

```suggestion
STDLIB_NAPI_MODULE_EXPORT_FCN( addon )

```

File: lib/node_modules/@stdlib/stats/base/sstdevyc/src/addon.c [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+* @param env    environment under which the function is invoked
+* @param info   callback data
+* @return       Node-API value
+*/
+static napi_value addon( napi_env env, napi_callback_info info ) {
+	STDLIB_NAPI_ARGV( env, info, argv, argc, 4 );
+	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
+	STDLIB_NAPI_ARGV_FLOAT( env, correction, argv, 1 );
+	STDLIB_NAPI_ARGV_INT64( env, stride, argv, 3 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, stride, argv, 2)
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/sstdevyc/src/addon.c]:
```suggestion
	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, stride, argv, 2 )
```

**Review by kgryte — APPROVED** [review]
LGTM.

File: lib/node_modules/@stdlib/stats/base/sstdevyc/src/addon.c (hunk 6 above)
Comment (Vinit-Pandit) [reply lib/node_modules/@stdlib/stats/base/sstdevyc/src/addon.c]: Reputation  < 0.   🥲

**Pull request merged**